src/components/Modal/VotingPowerDelegationDetail/VotingPowerDistribution.tsx

```diff
@@ -28,106 +31,134 @@ const VotingPowerDistribution = ({ vpDistribution, className }: Props) => {
-            label={t('modal.vp_delegation.details.stats_mana')}
-          >
-            <div
-              className="VotingPowerDistributionBar__Item VotingPowerDistribution__Mana"
-              style={{ width: manaPercentage }}
-            />
-          </VotingPowerDistributionPopup>
-        )}
-        {names > 0 && (
-          <VotingPowerDistributionPopup
-            amount={names}
-            percentage={namePercentage}
-            label={t('modal.vp_delegation.details.stats_name')}
-          >
-            <div
-              className="VotingPowerDistributionBar__Item VotingPowerDistribution__Name"
-              style={{ width: namePercentage }}
-            />
-          </VotingPowerDistributionPopup>
-        )}
-        {linkedWearables > 0 && (
-          <VotingPowerDistributionPopup
-            amount={linkedWearables}
-            percentage={linkedWearablesPercentage}
-            label={t('modal.vp_delegation.details.stats_linked_wearables')}
-          >
-            <div
-              className="VotingPowerDistributionBar__Item VotingPowerDistribution__LinkedWearables"
-              style={{ width: linkedWearablesPercentage }}
-            />
-          </VotingPowerDistributionPopup>
-        )}
-        {land > 0 && (
-          <VotingPowerDistributionPopup
-            amount={land}
-            percentage={landPercentage}
-            label={t('modal.vp_delegation.details.stats_land')}
-          >
-            <div
-              className="VotingPowerDistributionBar__Item VotingPowerDistribution__Land"
-              style={{ width: landPercentage }}
-            />
-          </VotingPowerDistributionPopup>
-        )}
-        {estate > 0 && (
-          <VotingPowerDistributionPopup
-            amount={estate}
-            percentage={estatesPercentage}
-            label={t('modal.vp_delegation.details.stats_estate')}
-          >
-            <div
-              className="VotingPowerDistributionBar__Item VotingPowerDistribution__Estate"
-              style={{ width: estatesPercentage }}
-            />
-          </VotingPowerDistributionPopup>
-        )}
-        {delegated > 0 && (
-          <VotingPowerDistributionPopup
-            amount={delegated}
-            percentage={delegatedPercentage}
-            label={t('modal.vp_delegation.details.stats_delegated')}
-          >
-            <div
-              className="VotingPowerDistributionBar__Item VotingPowerDistribution__Delegated"
-              style={{ width: delegatedPercentage }}
-            />
-          </VotingPowerDistributionPopup>
-        )}
-      </div>
+      {isLoading ? (
+        <Skeleton className={TokenList.join(['VotingPowerDistributionBar', 'VotingPowerDistributionBar__Loading'])} />
+      ) : (
+        <div
+          className={TokenList.join(['VotingPowerDistributionBar', total <= 0 && 'VotingPowerDistributionBar--Empty'])}
+        >
+          {mana > 0 && (
+            <VotingPowerDistributionPopup
```

Comment:
I feel like this section could be perfomed with a for loop, WDYT? a lot of repetition

Reply:
How about now?
A for loop seems tricky because you'd need to create a sort of mapping from variables to css classes and labels, or use Object.keys and loop. But I don't like so much the Object.keys approach because if it changes the order or a variable name, or you add something new to it, it breaks.

Not sure if the added complexity is worth in this case...